src/sync/sync_manager.cpp

```diff
@@ -168,6 +161,11 @@ void SyncManager::configure(SyncClientConfig config, util::Optional<app::App::Co
             m_users.emplace_back(std::move(user));
         }
     }
+
+    // App must be created last as the constructor depends on the SyncFileManager and
```

Comment:
The weird thing being done with make_scope_exit was because we need to create the App even if we return early due to not having metadata storage enabled, so this change breaks all the tests where we have it disabled.

I think it'd be clearer if the metadata storage init was extracted to a separate function that's conditionally called rather than the previous thing of using a scope exit.